components/dashboard/src/workspaces/WorkspaceEntry.tsx

```diff
@@ -111,18 +143,32 @@ export function WorkspaceEntry({ desc, model, isAdmin, stopWorkspace }: Props) {
             </Tooltip>
         </ItemField>
         <ItemFieldContextMenu menuEntries={menuEntries} />
-        {isModalVisible && <ConfirmationModal
+        {isDeleteModalVisible && <ConfirmationModal
             title="Delete Workspace"
             areYouSureText="Are you sure you want to delete this workspace?"
             children={{
                 name: ws.id,
                 description: ws.description,
             }}
             buttonText="Delete Workspace"
-            visible={isModalVisible}
-            onClose={() => setModalVisible(false)}
+            visible={isDeleteModalVisible}
+            onClose={() => setDeleteModalVisible(false)}
             onConfirm={() => model.deleteWorkspace(ws.id)}
         />}
+        <Modal visible={isRenameModalVisible} onClose={() => setRenameModalVisible(false)} onEnter={() => { updateWorkspaceDescription(); return true }}>
+            <h3 className="mb-4">Rename Workspace Description</h3>
+            <div className="border-t border-b border-gray-200 dark:border-gray-800 -mx-6 px-6 py-4 space-y-2">
+                <input className="w-full" type="text" defaultValue={workspaceDescription} ref={renameInputRef} />
```

Comment:
**issue**: Long descriptions seems to bring back a similar overflow issue as in https://github.com/gitpod-io/gitpod/issues/4556.

| Overflowing text when deleting a workspace |
|-|
| <img width="1440" alt="Screenshot 2021-09-15 at 7 36 55 PM (2)" src="https://user-images.githubusercontent.com/120486/133474463-729de495-75b5-4fd8-a86f-2e24aa704dde.png"> |

Reply:
We should truncate it I suppose?

Reply:
I'd vote for truncation to also avoid breaking this component!

On the other side, we could go all-in by a) wrapping the text, b) allowing longer description to surface on the dashboard, etc. _but_ this would be against the notion of ephemeral environments. Maybe this is something to consider in future if needed. 🗳️

Reply:
Pushed a solution to show an error message and also truncating it. Let me know of a better error message! I'm off tomorrow, so feel free to change it here as well if I don't see the follow-up comment today.
<img width="488" alt="Screenshot 2021-09-16 at 15 19 18" src="https://user-images.githubusercontent.com/8015191/133628950-e4e9857e-7be1-42ab-ba3c-4d4f757ab72c.png">